lib/block-patterns.php

```diff
@@ -302,3 +302,134 @@ function( $current_screen ) {
 		}
 	}
 );
+
+/**
+ * FIXME.
+ *
+ * @since 5.9.0
+ */
+function _register_theme_block_patterns() {
+	$default_headers = array(
+		'title'         => 'Pattern Name',
+		'description'   => 'Description',
+		'viewportWidth' => 'Viewport Width',
+		'categories'    => 'Categories',
+		'keywords'      => 'Keywords',
+		'blockTypes'    => 'Block Types',
+	);
+
+	// Register patterns for the active theme, for both parent and child theme,
+	// if applicable.
+	foreach ( wp_get_active_and_valid_themes() as $theme ) {
+		$dirpath = $theme . '/patterns/';
```

Comment:
Any particular reason not to keep it backward-compatible? We could still support block patterns as we do today while allowing pure PHP and HTML files to describe them. We'd just need to filter the files using the "contain metadata" predicate.

Reply:
If I understood your question correctly, this is already done a few lines below:

```php
// Title is a required property.
if ( ! $pattern_data['title'] ) {
	continue;
}
```